src/dlsproto/node/neo/internal/SuspendableRequest.d

```diff
@@ -194,13 +195,18 @@ public struct SuspendableRequest
         {
             send_interrupted = false;
 
-            this.conn.sendReceive(
-                ( in void[] msg ) {send_interrupted = true;},
+            auto event = this.conn.nextEvent(this.conn.NextEventFlags.Receive,
                 ( conn.Payload payload )
                 {
                     payload.add(channel_removed_msg);
                 }
             );
+
+            if (event.active == event.active.received)
```

Comment:
This probably warrants a comment. The code appears to be registering to read but then ignoring any messages that are received.

Reply:
:+1: I'll add the comment.

Reply:
Comment.

Reply:
Probably should assert / document the fiber resumed case.